src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -333,7 +333,6 @@ namespace winrt::Microsoft::Terminal::Control::implementation
             // (The window has a min. size that ensures that there's always a scrollbar thumb.)
             if (drawableRange < 0)
             {
-                assert(false);

```

Comment:
What's wrong with this assertion?

Reply:
This has hit for me literally all the time in debug builds. I've got marks set up in my prompt, so I get this on a scrollbar update all the time on startup, after initialization but before the window's actually been drawn. Usually I just have to delete this line and I re-add it before I commit & push, I forgot this time

Reply:
Is there a way to early-return somewhere further up in the function in this case? For instance, would replacing 299 to 303 with this work?

```cpp
const auto scrollBarWidthInDIP = scrollBar.ActualWidth();
const auto scrollBarHeightInDIP = scrollBar.ActualHeight();

// Under certain circumstances (= during startup, after initialization and before the first WinUI layout),
// the scrollbar may have empty dimensions. This uses inverted checks to protect against NaN just in case.
if (!(scrollBarWidthInDIP > 0) || !(scrollBarHeightInDIP > 0)) {
    return;
}

const auto scaleFactor = DisplayInformation::GetForCurrentView().RawPixelsPerViewPixel();
const auto scrollBarWidthInPx = gsl::narrow_cast<int32_t>(lrint(scrollBarWidthInDIP * scaleFactor));
const auto scrollBarHeightInPx = gsl::narrow_cast<int32_t>(lrint(scrollBarHeightInDIP * scaleFactor));
// ...
```

Reply:
The reason I'd like to keep the assertion is so that we still have a debug-check for this subtraction:
```cpp
const auto drawableRange = scrollBarHeightInPx - 2 * increaseDecreaseButtonHeight;
```

Reply:
well tho

![image](https://github.com/microsoft/terminal/assets/18356694/87a96743-8fac-4415-b355-7a1ed48e5c87)

Reply:
Wat.

Hmm, ok, nevermind then. 😅

Reply:
WHY IS THIS IN EVERY PR LOL

Reply:
use `git add -p` to individually consider which hunks belong on each branch or in each commit ;)